src/crimson/osd/pg.h

```diff
@@ -797,6 +800,7 @@ class PG : public boost::intrusive_ref_counter<
   std::map<ceph_tid_t, log_update_t> log_entry_update_waiting_on;
   // snap trimming
   interval_set<snapid_t> snap_trimq;
+  bool snap_trimming = false;
```

Comment:
Why not re-use `PG_STATE_SNAPTRIM`? e.g `peering_state.state_test(PG_STATE_SNAPTRIM)`

Reply:
Um, it seems that we need to prevent PG from creating extra `SnapTrimEvent`s when there's already one; on the other hand, `PG_STATE_SNAPTRIM` is set after `reservations` are done, which means there's a time interval between the creation of `SnapTrimEvent` and the set of `PG_STATE_SNAPTRIM`.